Sources/KituraNet/BufferList.swift

```diff
@@ -147,7 +147,12 @@ public class BufferList {
 
     */
     public func fill(array: inout [UInt8]) -> Int {
-        return fill(buffer: UnsafeMutablePointer(mutating: array), length: array.count)
+        var filledBytes = 0
+        let arrayCount = array.count
+        array.withUnsafeMutableBufferPointer({ bufferPointer in
+            filledBytes = fill(buffer: bufferPointer.baseAddress!, length: arrayCount)
```

Comment:
```suggestion
            filledBytes = fill(buffer: bufferPointer.baseAddress!, length: bufferPointer.count)
```

you shouldn't really refer to `array` in here

Reply:
Hi @weissi thanks for helping to review this PR.  Can you suggest a better way to accomplish this?  Based on the description in the comments the goal of this function is to fill `array` with the contents of the bufferList so I must refer to `array` in some way.  Unfortunately with the new `withUnsafeMutableBufferPointer` form, I cannot refer to `array.count` directly in the block, hence the local variable to keep track of the length of array.

Reply:
@dannys42 but `bufferPointer.count` will have the same value as `array.count` and you should be using that one.

However, two observations:
- It should never be necessary to use `*Unsafe*` stuff in Kitura-NIO (if it is, please file a NIO bug). And in this case, the easiest way to write the code you have there would probably be:

```swift
    public func fill(array: inout [UInt8]) -> Int {
        // I don't know why this API doesn't take all the available bytes, but let's stick to its contract.
        let bytesToFill = min(array.count, self.byteBuffer.readableBytes)

        // Get a `ByteBufferView` (a `Collection`) of all readable bytes.
        let readableByteView = self.byteBuffer.readableBytesView

        // We want to overwrite the first `bytesToFill` bytes with the first `bytesToFill` bytes from the `ByteBufferView`.
        array.replaceSubrange(0 ..< bytesToFill,
                              with: readableByteView.prefix(bytesToFill))

        // And then move the reader index to indicate that we consumed the bytes.
        self.byteBuffer.moveReaderIndex(forwardBy: bytesToFill)

        return bytesToFill
    }
```

- is anybody actually using these APIs?

Reply:
@weissi thanks!  The Kitura-NIO package is attempting to be API compatible with KituraNet (which was created pre-NIO).  Right now I'm just trying to update the existing code to avoid the compiler warnings.  Since I'm not sure wasn't sure what the performance impact is of moving to pure swift functions, nor am I that familiar with NIO, I opted for making the smallest amount of changes.  But your changes definitely look clean, so I'm happy to incorporate them.

I'm not sure if there are other packages using this fill() API.  But KituraNet does appear to be using it for FastCGI support, which does not look like it's implemented yet in Kitura-NIO.  So it's possible it's not an issue at this time.  But I'll be updating KituraNet with similar changes in order to get rid of these warnings.

Reply:
FYI, update to Kitura-net is in this [PR](https://github.com/Kitura/Kitura-net/pull/321)

Reply:
Cool, sounds good! Regarding FastCGI: I would very much recommend to not support it (is there any point these days?) but if you wanted to support it, I'd transparently support it in Kitura-NIO.